airwallex/src/main/java/com/airwallex/android/AirwallexStarter.kt

```diff
@@ -132,101 +125,99 @@ class AirwallexStarter {
 
-                        PaymentMethodsActivityLaunch.REQUEST_CODE -> {
-                            val result =
-                                PaymentMethodsActivityLaunch.Result.fromIntent(data) ?: return true
-                            when {
-                                result.exception != null -> {
-                                    paymentResultListener?.onCompleted(
-                                        AirwallexPaymentStatus.Failure(result.exception)
-                                    )
-                                }
-
-                                result.paymentIntentId != null -> {
-                                    if (result.isRedirecting) {
-                                        paymentResultListener?.onCompleted(
-                                            AirwallexPaymentStatus.InProgress(result.paymentIntentId)
-                                        )
-                                    } else {
-                                        paymentResultListener?.onCompleted(
-                                            AirwallexPaymentStatus.Success(result.paymentIntentId)
-                                        )
-                                    }
-                                }
+                        result.paymentIntentId != null -> {
+                            if (result.isRedirecting) {
+                                paymentResultListener.onCompleted(
+                                    AirwallexPaymentStatus.InProgress(result.paymentIntentId)
+                                )
+                            } else {
+                                paymentResultListener.onCompleted(
+                                    AirwallexPaymentStatus.Success(result.paymentIntentId)
+                                )
                             }
-                            paymentResultListener = null
-                            true
                         }
-
-                        else -> false
                     }
                 }
 
                 Activity.RESULT_CANCELED -> {
-                    return when (requestCode) {
-                        PaymentShippingActivityLaunch.REQUEST_CODE -> {
-                            shippingResultListener?.onCompleted(AirwallexShippingStatus.Cancel)
-                            shippingResultListener = null
-                            true
-                        }
-
-                        PaymentMethodsActivityLaunch.REQUEST_CODE -> {
-                            paymentResultListener?.onCompleted(AirwallexPaymentStatus.Cancel)
-                            paymentResultListener = null
-                            true
-                        }
-
-                        else -> false
-                    }
+                    paymentResultListener.onCompleted(AirwallexPaymentStatus.Cancel)
                 }
-
-                else -> return false
             }
         }
     }
+
+    class PaymentException(
```

Comment:
Can we reuse `AirwallexCheckoutException` which has same effect?

Reply:
ok, I will replace it